### ISSUE TITLE
Deprecated parameters make the server always return 422 response

### ISSUE THREAD

**Issue by lehphyro** [stoplightio/prism#714]
**Describe the bug**
I cannot get a successful response if my spec has a deprecated parameter.

**To Reproduce**

1. Given this OpenAPI document:
```openapi: 3.0.1
info:
  title: Test
  version: v1
paths:
  /test:
    description: Test request
    get:
      summary: Summary
      operationId: getTest
      parameters:
      - deprecated: true
        description: Deprecated parameter
        example: 11-22-33
        in: query
        name: productId
        required: false
        schema:
          type: string
      responses:
        200:
          description: successful response
          content:
            application/json:
              schema:
                type: object
                properties:
                  prop1:
                    description: Test prod
                    type: string
```
2. Run this CLI command: `prism mock openapi.yaml`
3. See error:
```
{
    "type": "https://stoplight.io/prism/errors#UNPROCESSABLE_ENTITY",
    "title": "Invalid request body payload",
    "status": 422,
    "detail": "Your request body is not valid and no HTTP validation response was found in the spec, so Prism is generating this error for you.",
    "validation": [
        {
            "location": [
                "query",
                "productId"
            ],
            "severity": "Warning",
            "code": "deprecated",
            "message": "Query param productId is deprecated"
        }
    ]
}
```

**Expected behavior**
Make a `GET http://localhost:4010/test` request and get a `200 OK` response.

**Environment (remove any that are not applicable):**
 - Library version: 3.1.0

**Additional context**
Removing the `deprecated` flag makes it work correctly.

**Comment by XVincentX** [comment]
Good point. A deprecated parameter does not make the request invalid. 

Still, we want to be able to report this thing accordingly somehow. I'll try to think about it and see how we can have the cake and eat it too.

**Comment by XVincentX** [comment]
@lag-of-death You can see that this particular error (causing the issue) is already marked as a warning and not as an error https://github.com/stoplightio/prism/blob/master/packages/http/src/validator/validators/params.ts#L25

For this reason, we need to modify the negotiator to trigger the negotiation for an invalid request only if there's at least an error.

https://github.com/stoplightio/prism/blob/master/packages/http/src/mocker/index.ts#L92

That should do the trick. We will also need to log these warnings (you should have the logger interface already injected, so that should be easy)

Add tests, and we're good to go.

**Comment by XVincentX** [comment]
We have just released 3.2.0-beta.5 that fixes this issue. You can try it out installing Prism from npm with its beta tag

`npm i @stoplight/prism-cli@beta`

**Comment by XVincentX** [comment]
@lehphyro Prism 3.2.0's been released!
